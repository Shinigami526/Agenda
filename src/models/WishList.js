import { types, getParent, destroy } from 'mobx-state-tree'

export const WishListItem = types.model({
    name: types.string,
    number: types.string,
    image: types.string,
    summary: types.string,
    text: types.string
})
  .actions(self => ({
      changeName(newName) {
          self.name = newName
      },
      changeNumber(newPrice) {
          self.number = newPrice
      },
      changeImage(newImage) {
          self.image = newImage
      },
      changeSummary(newSummary){
          self.summary = newSummary
      },
      changeText(newText){
          self.text = newText
      },
      remove(){
         getParent(self, 2).remove(self)
      }
  }))

export const Wishlist = types.model({
    items: types.optional(types.array(WishListItem), [])
})
  .actions(self => ({
      add(item) {
          self.items.push(item)
      },
      remove(item) {
        destroy(item)
      }
  }))
     