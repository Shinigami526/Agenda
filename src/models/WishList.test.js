import { WishListItem, Wishlist } from './WishList'
import { getSnapshot, onSnapshot } from 'mobx-state-tree'

it('can create a instance of a model', () => {
    const item = WishListItem.create({
        "name": "Chronicles of Narnia Box Set - C.S Lewis",
        "number": "0111565471",
        "image": "",
        "summary":"Chronicles of Narnia",
        "text": "Box Set - C.S Lewis" 
    })

    expect(item.number).toBe('0111565471')
    expect(item.image).toBe("")
    item.changeName("Narnia")
    expect(item.name).toBe("Narnia")
})


it("can add new items", () => {
    const list = Wishlist.create()
    const states = []
    onSnapshot(list, snapshot => {
        states.push(snapshot)
    })
    list.add({
        name: "Chesterton",
        number: "0111565225",
        image: "",
        summary: "Chronicles of Narnia",
        text: "Box Set - C.S Lewis"
    })
    expect(list.items.length).toBe(1)
    expect(list.items[0].name).toBe('Chesterton')
    list.items[0].changeName("Book of G.K Chesterton")
    expect(list.items[0].name).toBe("Book of G.K Chesterton")
    
    expect(getSnapshot(list)).toMatchSnapshot()
    
    expect(states).toMatchSnapshot()
})

