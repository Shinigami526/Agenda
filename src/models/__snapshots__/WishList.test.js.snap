// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new items 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "Book of G.K Chesterton",
      "number": "0111565225",
      "summary": "Chronicles of Narnia",
      "text": "Box Set - C.S Lewis",
    },
  ],
}
`;

exports[`can add new items 2`] = `
Array [
  Object {
    "items": Array [
      Object {
        "image": "",
        "name": "Chesterton",
        "number": "0111565225",
        "summary": "Chronicles of Narnia",
        "text": "Box Set - C.S Lewis",
      },
    ],
  },
  Object {
    "items": Array [
      Object {
        "image": "",
        "name": "Book of G.K Chesterton",
        "number": "0111565225",
        "summary": "Chronicles of Narnia",
        "text": "Box Set - C.S Lewis",
      },
    ],
  },
]
`;
